src/MIDebugEngine/Engine.Impl/Disassembly.cs

```diff
@@ -166,10 +143,123 @@ public Disassembly(DebuggedProcess process)
                 }
                 var kv = block.TryFetch(address, nInstructions, out ret);
             }
-
             return ret;
         }
 
+        /// <summary>
+        /// Seek backward n instructions from a target address
+        /// </summary>
+        /// <param name="address">target address</param>
+        /// <param name="nInstructions">number of instructions</param>
+        /// <returns> address - n on failure, else the address of an instruction n back from the target address</returns>
+        public async Task<ulong> SeekBack(ulong address, int nInstructions)
+        {
+            ICollection<DisasmInstruction> ret = null;
+            ulong defaultAddr = address >= (ulong)nInstructions ? address - (ulong)nInstructions : 0;
+
+            lock (_disassemlyCache)
+            {
+                // check the cache
+                var kv = _disassemlyCache.FirstOrDefault((p) => p.Value.TryFetch(address, -nInstructions, out ret));
+                if (kv.Value != null)
+                    return ret.First().Addr;
```

Comment:
This will seek back nInstruction bytes. But if the count of instructions is less than nInstructions, don't we need to keep looking?